### DISCUSSION TITLE
Does nodejs core still support xcode-debugging on MacOS?

### DISCUSSION THREAD

**Discussion by zhuyingda** [nodejs/help#4925] [Q&A]
### What is the problem this feature will solve?

I remember it is supported some years ago that I can debug nodejs core (c++ code) by MacOS xcode after `configure --xcode`.
Now, it seems like I can not do this anymore.
I have read the [document of building](https://github.com/nodejs/node/blob/master/BUILDING.md) and found nothing about this.
I would appreciate if somebody could tell me how to debug the nodejs core on MacOS platform. Thanks.

### What is the feature you are proposing to solve the problem?

debug nodejs core on MacOS platform.
:)

### What alternatives have you considered?

Or please tell me how to debug Node.js on other platform? such as Linux...I have to build up my environment.

**Comment by targos** [comment]
The flag was removed a long time ago. See https://github.com/nodejs/node/pull/20328#issuecomment-384645003 for a possible workaround.

**Comment by targos** [comment]
Actually, I'll keep this open. Maybe some people can point you to other ways to debug on macOS.

**Comment by zhuyingda** [comment]
> The flag was removed a long time ago. See [#20328 (comment)](https://github.com/nodejs/node/pull/20328#issuecomment-384645003) for a possible workaround.

Thanks for your reply, @targos  :)

**Comment by zhuyingda** [comment]
@targos by the way may I ask what develop platform are you using, Linux platform?
It seems like I am just minority user who use XCode IDE to debug C++ code...

**Comment by targos** [comment]
I use macOS, Linux and Windows but I don't know how to debug C++ code effectively. I have basic knowledge of `lldb` on macOS and `gdb` on Linux.

**Comment by zhuyingda** [comment]
> I use macOS, Linux and Windows but I don't know how to debug C++ code effectively. I have basic knowledge of `lldb` on macOS and `gdb` on Linux.

so，how do you often debug nodejs core？
maybe lldb on MacOS？

**Comment by bnoordhuis** [comment]
lldb is what I use on macOS (grudgingly, gdb muscle memory is too ingrained.)

**Comment by zhuyingda** [comment]
> I use macOS, Linux and Windows but I don't know how to debug C++ code effectively. I have basic knowledge of `lldb` on macOS and `gdb` on Linux.

@targos Do you use lldb on MacOS to debug nodejs core? :)

**Comment by RaisinTen** [comment]
Same here:
* lldb on macOS
* gdb on linux and
* stay away from Windows related things mostly :P. Maybe WinDBG would be useful on Windows but I haven't tried it yet.

**Comment by targos** [comment]
> > I use macOS, Linux and Windows but I don't know how to debug C++ code effectively. I have basic knowledge of `lldb` on macOS and `gdb` on Linux.
> 
> @targos Do you use lldb on MacOS to debug nodejs core? :)

Yes, but I don't really spend my time debugging node's C++ code

**Comment by zhuyingda** [comment]
> Same here:
> 
> * lldb on macOS
> * gdb on linux and
> * stay away from Windows related things mostly :P. Maybe WinDBG would be useful on Windows but I haven't tried it yet.

Thanks, and yes I do not like to do these things on Windows platform, too. :)

**Comment by zhuyingda** [comment]
> > > I use macOS, Linux and Windows but I don't know how to debug C++ code effectively. I have basic knowledge of `lldb` on macOS and `gdb` on Linux.
> > 
> > 
> > @targos Do you use lldb on MacOS to debug nodejs core? :)
> 
> Yes, but I don't really spend my time debugging node's C++ code

I think I should use lldb but not Xcode IDE, after all Xcode debugging is base on lldb.
Could you tell me any documentation about lldb debugging Nodejs core?

**Reply by bnoordhuis** [reply]
> Could you tell me any documentation about lldb debugging Nodejs core?

It's like debugging any other C++ program except JS stack frames confuse lldb due to their custom calling convention. [llnode](https://github.com/nodejs/llnode) can decode them though.

**Comment by targos** [comment]
I'm not the one you should ask those questions.

**Comment by himself65** [comment]
I usually debug with `CLion` on all platforms. This is my thread about how to start on macOS
https://twitter.com/himseIf_65/status/1517393226474303488

**Comment by pumpdog** [comment]
поддерживает
